core/src/main/java/io/aiven/klaw/service/OperationalRequestsService.java

```diff
@@ -0,0 +1,337 @@
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+public class OperationalRequestsService {
+
+  public static final String OFFSET_RESET_TIMESTAMP_FORMAT = "yyyy-MM-dd'T'HH:mm:ss.SSSZ";
+  private final ManageDatabase manageDatabase;
+  private final MailUtils mailService;
+  private final ClusterApiService clusterApiService;
+  private final CommonUtilsService commonUtilsService;
+  private final RolesPermissionsControllerService rolesPermissionsControllerService;
+
+  public OperationalRequestsService(
+      ManageDatabase manageDatabase,
+      MailUtils mailService,
+      ClusterApiService clusterApiService,
+      CommonUtilsService commonUtilsService,
+      RolesPermissionsControllerService rolesPermissionsControllerService) {
+    this.manageDatabase = manageDatabase;
+    this.mailService = mailService;
+    this.clusterApiService = clusterApiService;
+    this.commonUtilsService = commonUtilsService;
+    this.rolesPermissionsControllerService = rolesPermissionsControllerService;
+  }
+
+  public ApiResponse createConsumerOffsetsResetRequest(
+      ConsumerOffsetResetRequestModel consumerOffsetResetRequestModel)
+      throws KlawNotAuthorizedException {
+    log.info("createConsumerOffsetsResetRequest {}", consumerOffsetResetRequestModel);
+    checkIsAuthorized(PermissionType.REQUEST_CREATE_SUBSCRIPTIONS);
+    String userName = getUserName();
+    consumerOffsetResetRequestModel.setRequestor(userName);
+    consumerOffsetResetRequestModel.setRequestingTeamId(commonUtilsService.getTeamId(userName));
+
+    HandleDbRequests dbHandle = manageDatabase.getHandleDbRequests();
+
+    OperationalRequest operationalRequest = new OperationalRequest();
+    copyProperties(consumerOffsetResetRequestModel, operationalRequest);
+    operationalRequest.setTenantId(commonUtilsService.getTenantId(userName));
+    if (operationalRequest.getOffsetResetType() == OffsetResetType.TO_DATE_TIME) {
+      operationalRequest.setResetTimeStamp(
+          getUTCTimeStamp(consumerOffsetResetRequestModel.getResetTimeStampStr()));
+    }
+    String result = dbHandle.requestForConsumerOffsetsReset(operationalRequest).get("result");
+
+    String requestFormattedStr = "Consumer group : " + operationalRequest.getConsumerGroup();
+    mailService.sendMail(
+        operationalRequest.getTopicname(),
+        requestFormattedStr,
+        "",
+        operationalRequest.getRequestor(),
+        operationalRequest.getApprover(),
+        operationalRequest.getRequestingTeamId(),
+        dbHandle,
+        MailType.RESET_CONSUMER_OFFSET_REQUESTED,
+        commonUtilsService.getLoginUrl());
+
+    return ApiResultStatus.SUCCESS.value.equals(result)
+        ? ApiResponse.ok(result)
+        : ApiResponse.notOk(result);
+  }
+
+  private Timestamp getUTCTimeStamp(String timeStampStr) {
```

Comment:
Do we need to support non UTC Timestamps? Allow that the users can define the server time in the application.properties